eZ/Publish/Core/Base/Container/Compiler/Search/SignalSlotPass.php

```diff
@@ -20,31 +19,47 @@
  */
 class SignalSlotPass implements CompilerPassInterface
 {
+    /**
+     * Name of a search engine for which Compiler Pass was instantiated.
+     *
+     * @var string|null
+     */
+    private $searchEngineAlias;
+
+    public function __construct($searchEngineAlias = null)
+    {
+        $this->searchEngineAlias = $searchEngineAlias;
+    }
+
     public function process(ContainerBuilder $container)
     {
-        if (!$container->hasDefinition('ezpublish.signalslot.signal_dispatcher')) {
+        if (!$container->hasDefinition('ezpublish.signalslot.signal_dispatcher_factory')) {
             return;
         }
 
-        $signalDispatcherDef = $container->getDefinition('ezpublish.signalslot.signal_dispatcher');
+        $signalDispatcherDef = $container->getDefinition('ezpublish.signalslot.signal_dispatcher.factory');
+        $searchEngineSlotTagName = sprintf('ezpublish.search.%s.slot', $this->searchEngineAlias);
+        $tags = $container->findTaggedServiceIds('ezpublish.search.slot')
+                + $container->findTaggedServiceIds($searchEngineSlotTagName);
 
-        foreach ($container->findTaggedServiceIds('ezpublish.search.slot') as $id => $attributes) {
+        $dispatcherCallsArguments = [];
+        foreach ($tags as $id => $attributes) {
             foreach ($attributes as $attribute) {
                 if (!isset($attribute['signal'])) {
                     throw new LogicException(
                         "Could not find 'signal' attribute on '$id' service, " .
-                        "which is mandatory for services tagged as 'ezpublish.persistence.solr.slot'"
+                        "which is mandatory for services tagged as '$searchEngineSlotTagName'"
                     );
                 }
 
-                $signalDispatcherDef->addMethodCall(
-                    'attach',
-                    array(
-                        $attribute['signal'],
-                        new Reference($id),
-                    )
-                );
+                $dispatcherCallsArguments[] = new SignalDispatcherCallArguments([
+                    'searchEngineName' => $this->searchEngineAlias,
+                    'signal' => $attribute['signal'],
+                    'slot' => new Reference($id),
+                ]);
             }
         }
+
+        $signalDispatcherDef->addMethodCall('attachAll', $dispatcherCallsArguments);
```

Comment:
good, but could we inject in constructor instead somehow _(makes the factory cleaner)_? /cc @pspanja